xpdacq/xpdacq.py

```diff
@@ -192,6 +192,58 @@ def _parse_calibration_file(config_file_name):
                 config_dict[option] = None
     return config_dict
 
+def new_prun(sample, scanplan, auto_dark = glbl.auto_dark, **kwargs):
+    '''on this 'sample' run this 'scanplan'
+        
+    Arguments:
+    sample - sample metadata object
+    scanplan - scanplan metadata object
+    auto_dark - optional. Type auto_dark = False to suppress the automatic collection of a dark image (default = True). Strongly recommend to leave as True unless problems are encountered
+    **kwargs - dictionary that will be passed through to the run-engine metadata
+    '''
+    scan = Scan(sample, scanplan)
+    scan.md.update({'sc_isprun':True})
+    if auto_dark:
+        auto_dark_md_dict = _auto_dark(scan)
```

Comment:
Do you mean `auto_dark` and `_auto_dark` ?  `_auto_dark` is a function that executes automated dark collection and `auto_dark` is an option here. I think I'd better change function name cause it is even very confusing for me by just looking at it.

I am still working on this PR so I will correct it in following commits.